src/Main.elm

```diff
@@ -1,71 +1,392 @@
 module Main exposing (main)
 
 import Browser
-import Html exposing (..)
+import Browser.Dom as Dom
+import Css exposing (..)
+import Html
+import Html.Styled exposing (..)
+import Html.Styled.Attributes as HSA exposing (..)
+import Html.Styled.Events exposing (keyCode, on, onBlur, onCheck, onClick, onInput)
+import Http
+import HttpBuilder
+import Json.Decode as Json exposing (bool, field, int, string)
+import Json.Encode as Encode
+import Task
 
 
 
 -- MODEL
 
 
 type alias Todo =
-    { title : String }
+    { id : Int
+    , title : String
+    , completed : Bool
+    , url : String
+    }
 
 
 type alias Model =
-    { todos : List Todo }
+    { newTodoTitle : String
+    , todos : List Todo
+    , hideCompleted : Bool
+    , todoBeingEdited : Int
+    }
 
 
-initialModel : Model
-initialModel =
-    { todos =
-        [ { title = "Buy milk" }
-        , { title = "Wash dishes" }
-        ]
-    }
+type KeyDownWhere
+    = OnAdd
+    | OnRename
```

Comment:
Tare asta, de unde ti-a venit ideea?

Reply:
Aveam nevoie sa diferentiez, and this was the first thing that came into my mind.